### ISSUE TITLE
Support Structured Haskell Mode faces

### ISSUE THREAD

**Issue by geraldus** [cryon/subatomic#5]
I see you support a bunch of packages. Can you add support for SHM?

I've forked your repo to make a PR, but then decided to open new issue so you can tweak things a bit.

SHM have two face variables: `shm-current-face` and `shm-quarantine-face`. First one highlights current code around point and quarantine face is used to highlight broken code. I've choose these colours:

```elisp
…
   `(shm-current-face
     ((t (:background , "#34465e"))))
   `(shm-quarantine-face
     ((t (:background , "#421d17"))))
…
```
![2015-03-24 9 29 35](https://cloud.githubusercontent.com/assets/744471/6795826/79d4cf30-d208-11e4-8a57-6f8dbeb03496.png)

But  I suppose current face should be tweaked in high contrast mode, also your may prefer to change colours.

**Comment by cryon** [comment]
I'm always in favor of adding support for more modes!

I haven't tried SHM myself so I have no strong feelings about the colors for it. Though I would like to restrict the colors to the "[palette](https://github.com/cryon/subatomic/blob/master/subatomic-theme.el#L45-L69)". 

Perhaps setting the background of shm-current-face to a darker shade, like midnight-{1, 2, 3} and underline the shm-quarantine-face with either "full-red" or "brick-red" (both shades of red might be a bit too strong for a background color). Just a suggestion.

What do you think? Feel free to send me a PR with whatever you think is good!

PS. Would be awesome if your additions follow my style when it comes to location of "," and newlines. Example (untested code):
```elisp
   ;; structured-haskell-mode
   
   `(shm-current-face
     ((t (:background ,midnight-2))))
   
   `(shm-quarantine-face
     ((t (:underline ,brick-red))))
```

**Comment by geraldus** [comment]
> to restrict the colors to the "palette".

That's exactly why I opened this issue. (:
Well, `midnight-3` suits very well for `smh-current-face`, however, I can't decide what colour to pick up for quarantine face: full-red is too distracting for me; brick-red makes code hard to read because it's bright and `victory-blue` is blends with it. I guess some dark reddish colour will suit well, however there is no such colour in palette. So, I've defined new colour in palette. Sent PR.

Cheers!
